loadtest/user/userentity/actions.go

```diff
@@ -11,28 +11,120 @@ import (
 	"fmt"
 	"io"
 	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"regexp"
 	"strconv"
 
 	"github.com/graph-gophers/graphql-go"
 	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
+// Possible actions for the openID authentication
+type authOpenIDAction string
+
+const (
+	authOpenIDLogin  authOpenIDAction = "login"
+	authOpenIDSignup authOpenIDAction = "signup"
+)
+
+var (
+	openIDLoginFormActionRegex = regexp.MustCompile(`action=["'](.*?)["']`)
+)
+
 // SignUp signs up the user with the given credentials.
 func (ue *UserEntity) SignUp(email, username, password string) error {
-	user := model.User{
-		Email:    email,
-		Username: username,
-		Password: password,
+	var newUser *model.User
+	var err error
+
+	switch ue.config.AuthenticationType {
+	case AuthenticationTypeOpenID:
+		if err := ue.authOpenID(authOpenIDSignup); err != nil {
+			return fmt.Errorf("error while signing up using OpenID: %w", err)
+		}
+
+		newUser, _, err = ue.client.GetUserByUsername(context.Background(), username, "")
+		if err != nil {
+			return fmt.Errorf("error while getting user by username: %w", err)
+		}
+	default:
+		user := model.User{
+			Email:    email,
+			Username: username,
+			Password: password,
+		}
+
+		newUser, _, err = ue.client.CreateUser(context.Background(), &user)
+		if err != nil {
+			return err
+		}
+
+		newUser.Password = password
 	}
+	return ue.store.SetUser(newUser)
+}
 
-	newUser, _, err := ue.client.CreateUser(context.Background(), &user)
+// authOpenID logs the user in using OpenID.
+func (ue *UserEntity) authOpenID(action authOpenIDAction) error {
+	if action != authOpenIDLogin && action != authOpenIDSignup {
+		return errors.New("invalid openid auth action")
+	}
+
+	jar, err := cookiejar.New(nil)
 	if err != nil {
-		return err
+		return fmt.Errorf("error while creating cookie jar: %w", err)
 	}
 
-	newUser.Password = password
-	return ue.store.SetUser(newUser)
+	client := &http.Client{
+		Jar: jar,
+	}
+
+	// make a request to the openid login page of mattermost
+	resp, err := client.Get(ue.client.URL + "/oauth/openid/" + string(action))
```

Comment:
Should this be login/signup?
```suggestion
	// make a request to the openid login/signup page of mattermost
	resp, err := client.Get(ue.client.URL + "/oauth/openid/" + string(action))
```